src/pathfinder.rs

```diff
@@ -0,0 +1,266 @@
+use lazy_static::lazy_static;
+use mut_static::MutStatic;
+use num::integer::sqrt;
+use pathfinding::prelude::astar;
+use serde::{Deserialize, Serialize};
+use std::num::ParseIntError;
+
+#[derive(Serialize, Deserialize, Default, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+struct Node {
+    unique_id: usize,
+    x: usize,
+    y: usize,
+    z: usize,
+    connected_nodes_id: Vec<usize>,
+}
+
+impl Node {
+    fn successors(&self) -> Vec<(Self, usize)> {
+        self.connected_nodes_id
+            .iter()
+            .filter_map(|id| NODES.read().unwrap().get(*id).cloned())
```

Comment:
By the way, this constant cloning is terrible but I'm not sure how to avoid it. Using rc ?